File: src/main/java/fm/comunas/ddd/oe/adapter/api/jsonapi/dto/ObjUserDto.java
package fm.comunas.ddd.oe.adapter.api.jsonapi.dto;

import io.crnk.core.resource.annotations.JsonApiId;
import fm.comunas.ddd.oe.model.ObjUser;
import lombok.Builder;
import lombok.Data;

@Data()
@Builder
public class ObjUserDto {

	@JsonApiId
	private Integer id;
	private String caption;
	private String name;
	private String email;
	private String picture;

	public static ObjUserDto fromObj(ObjUser obj) {
		if (obj == null) {
			return null;
		}
		// @formatter:off
		return ObjUserDto.builder()
			.id(obj.getId())
			.caption(obj.getCaption())
			.name(obj.getName())
			.email(obj.getEmail())
			.picture(obj.getPicture())
			.build();
		// @formatter:on
	}

}

File: fm-domain/src/main/java/io/zeitwert/fm/oe/adapter/api/jsonapi/dto/ObjUserDto.kt
package io.zeitwert.fm.oe.adapter.api.jsonapi.dto

import io.crnk.core.resource.annotations.JsonApiResource
import io.zeitwert.fm.obj.adapter.api.jsonapi.base.ObjDtoBase

@JsonApiResource(type = "user", resourcePath = "oe/users")
class ObjUserDto : ObjDtoBase()
